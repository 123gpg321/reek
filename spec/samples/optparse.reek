[Large Class] AmbiguousArgument has 52 methods
[Long Method] AmbiguousArgument#options has approx 6 statements
[Long Parameter List] CompletingHash#complete has 4 parameters
[Nested Iterators] CompletingHash#complete has nested iterators
[Feature Envy] CompletingHash#environment uses :env more than self
[Long Method] CompletingHash#getopts has approx 17 statements
[Long Method] CompletingHash#initialize has approx 8 statements
[Feature Envy] CompletingHash#load uses :File and :filename more than self
[Long Method] CompletingHash#make_switch has approx 69 statements
[Nested Iterators] CompletingHash#make_switch has nested iterators
[Uncommunicative Name] CompletingHash#make_switch uses the local variable name 'a'
[Uncommunicative Name] CompletingHash#make_switch uses the local variable name 'n'
[Uncommunicative Name] CompletingHash#make_switch uses the local variable name 'o'
[Uncommunicative Name] CompletingHash#make_switch uses the local variable name 'q'
[Uncommunicative Name] CompletingHash#make_switch uses the local variable name 's'
[Nested Iterators] CompletingHash#match has nested iterators
[Feature Envy] CompletingHash#order uses :argv more than self
[Feature Envy] CompletingHash#parse uses :argv more than self
[Long Method] CompletingHash#parse_in_order has approx 31 statements
[Nested Iterators] CompletingHash#parse_in_order has nested iterators
[Feature Envy] CompletingHash#permute uses :argv more than self
[Feature Envy] CompletingHash#permute! uses :arg more than self
[Long Parameter List] CompletingHash#summarize has 4 parameters
[Uncommunicative Name] CompletingHash#ver uses the local variable name 'v'
[Feature Envy] CompletingHash#visit uses :el more than self
[Feature Envy] List#accept uses :pat more than self
[Uncommunicative Name] List#accept uses the parameter name 't'
[Feature Envy] List#add_banner uses :opt more than self
[Long Parameter List] List#complete has 4 parameters
[Uncommunicative Name] List#reject uses the parameter name 't'
[Feature Envy] List#summarize uses :opt more than self
[Long Parameter List] List#update has 5 parameters
[Long Method] List#update has approx 6 statements
[Feature Envy] List#update uses :o more than self
[Uncommunicative Name] List#update uses the local variable name 'o'
[Long Method] OptionParser#complete has approx 23 statements
[Feature Envy] OptionParser#complete uses :candidates and :k more than self
[Uncommunicative Name] OptionParser#complete uses the local variable name 'k'
[Uncommunicative Name] OptionParser#complete uses the local variable name 'v'
[Utility Function] OptionParser#convert doesn't depend on instance state
[Long Method] PlacedArgument#parse has approx 6 statements
[Feature Envy] PlacedArgument#parse uses :val more than self
[Uncommunicative Name] Switch#add_banner uses the local variable name 's'
[Long Parameter List] Switch#initialize has 7 parameters
[Long Method] Switch#parse_arg has approx 12 statements
[Feature Envy] Switch#parse_arg uses :m and :s more than self
[Uncommunicative Name] Switch#parse_arg uses the local variable name 'm'
[Uncommunicative Name] Switch#parse_arg uses the local variable name 's'
[Long Parameter List] Switch#summarize has 5 parameters
[Long Method] Switch#summarize has approx 27 statements
[Nested Iterators] Switch#summarize has nested iterators
[Feature Envy] Switch#summarize uses :s more than self
[Uncommunicative Name] Switch#summarize uses the local variable name 'l'
[Uncommunicative Name] Switch#summarize uses the local variable name 'r'
[Uncommunicative Name] Switch#summarize uses the local variable name 's'
[Long Yield List] Switch#summarize yields 4 parameters
