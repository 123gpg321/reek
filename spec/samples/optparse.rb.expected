OptionParser has at least 60 methods (Large Class)
OptionParser#CompletingHash#match/block/block is nested (Nested Iterators)
OptionParser#Completion::complete calls candidates.size multiple times (Duplication)
OptionParser#Completion::complete calls k.id2name multiple times (Duplication)
OptionParser#Completion::complete has approx 23 statements (Long Method)
OptionParser#Completion::complete has the variable name 'k' (Uncommunicative Name)
OptionParser#Completion::complete has the variable name 'v' (Uncommunicative Name)
OptionParser#Completion::complete refers to candidates more than self (Feature Envy)
OptionParser#List#accept has the variable name 't' (Uncommunicative Name)
OptionParser#List#accept is controlled by argument pat (Control Couple)
OptionParser#List#accept refers to pat more than self (Feature Envy)
OptionParser#List#complete has 4 parameters (Long Parameter List)
OptionParser#List#reject has the variable name 't' (Uncommunicative Name)
OptionParser#List#update has 5 parameters (Long Parameter List)
OptionParser#List#update has approx 6 statements (Long Method)
OptionParser#List#update has the variable name 'o' (Uncommunicative Name)
OptionParser#List#update is controlled by argument lopts (Control Couple)
OptionParser#List#update is controlled by argument sopts (Control Couple)
OptionParser#ParseError#set_option is controlled by argument eq (Control Couple)
OptionParser#Switch#NoArgument#parse is controlled by argument arg (Control Couple)
OptionParser#Switch#OptionalArgument#parse is controlled by argument arg (Control Couple)
OptionParser#Switch#PlacedArgument#parse has approx 6 statements (Long Method)
OptionParser#Switch#RequiredArgument#parse is controlled by argument arg (Control Couple)
OptionParser#Switch#add_banner has the variable name 's' (Uncommunicative Name)
OptionParser#Switch#initialize has 7 parameters (Long Parameter List)
OptionParser#Switch#parse_arg calls s.length multiple times (Duplication)
OptionParser#Switch#parse_arg has approx 12 statements (Long Method)
OptionParser#Switch#parse_arg has the variable name 'm' (Uncommunicative Name)
OptionParser#Switch#parse_arg has the variable name 's' (Uncommunicative Name)
OptionParser#Switch#parse_arg refers to arg more than self (Feature Envy)
OptionParser#Switch#self.guess has the variable name 't' (Uncommunicative Name)
OptionParser#Switch#self.incompatible_argument_styles has the variable name 't' (Uncommunicative Name)
OptionParser#Switch#summarize calls block.max multiple times (Duplication)
OptionParser#Switch#summarize calls block.max.to_i multiple times (Duplication)
OptionParser#Switch#summarize calls indent.+(l) multiple times (Duplication)
OptionParser#Switch#summarize calls left.collect multiple times (Duplication)
OptionParser#Switch#summarize calls left.shift multiple times (Duplication)
OptionParser#Switch#summarize calls left[-1] multiple times (Duplication)
OptionParser#Switch#summarize calls s.length multiple times (Duplication)
OptionParser#Switch#summarize has 5 parameters (Long Parameter List)
OptionParser#Switch#summarize has approx 27 statements (Long Method)
OptionParser#Switch#summarize has the variable name 'l' (Uncommunicative Name)
OptionParser#Switch#summarize has the variable name 'r' (Uncommunicative Name)
OptionParser#Switch#summarize has the variable name 's' (Uncommunicative Name)
OptionParser#Switch#summarize refers to left more than self (Feature Envy)
OptionParser#Switch#summarize/block/block is nested (Nested Iterators)
OptionParser#block has the variable name 'f' (Uncommunicative Name)
OptionParser#block has the variable name 'k' (Uncommunicative Name)
OptionParser#block has the variable name 'o' (Uncommunicative Name)
OptionParser#block has the variable name 's' (Uncommunicative Name)
OptionParser#block/block has the variable name 'v' (Uncommunicative Name)
OptionParser#block/block is nested (Nested Iterators)
OptionParser#complete has 4 parameters (Long Parameter List)
OptionParser#complete/block/block is nested (Nested Iterators)
OptionParser#getopts has approx 17 statements (Long Method)
OptionParser#load/block has the variable name 's' (Uncommunicative Name)
OptionParser#make_switch calls default_style.guess(arg=a) multiple times (Duplication)
OptionParser#make_switch calls long.<<(o=q.downcase) multiple times (Duplication)
OptionParser#make_switch calls pattern.method(convert) multiple times (Duplication)
OptionParser#make_switch calls pattern.method(convert).to_proc multiple times (Duplication)
OptionParser#make_switch calls pattern.respond_to?(convert) multiple times (Duplication)
OptionParser#make_switch calls q.downcase multiple times (Duplication)
OptionParser#make_switch calls sdesc.<<("-#{q}") multiple times (Duplication)
OptionParser#make_switch has approx 69 statements (Long Method)
OptionParser#make_switch has the variable name 'a' (Uncommunicative Name)
OptionParser#make_switch has the variable name 'n' (Uncommunicative Name)
OptionParser#make_switch has the variable name 'o' (Uncommunicative Name)
OptionParser#make_switch has the variable name 'q' (Uncommunicative Name)
OptionParser#make_switch has the variable name 's' (Uncommunicative Name)
OptionParser#make_switch/block/block has the variable name 'c' (Uncommunicative Name)
OptionParser#make_switch/block/block has the variable name 'v' (Uncommunicative Name)
OptionParser#make_switch/block/block is nested (Nested Iterators)
OptionParser#make_switch/block/block/block is nested (Nested Iterators)
OptionParser#order calls argv[0] multiple times (Duplication)
OptionParser#order refers to argv more than self (Feature Envy)
OptionParser#parse calls argv[0] multiple times (Duplication)
OptionParser#parse refers to argv more than self (Feature Envy)
OptionParser#parse_in_order calls $!.set_option(arg, true) multiple times (Duplication)
OptionParser#parse_in_order calls cb.call(val) multiple times (Duplication)
OptionParser#parse_in_order calls setter.call(sw.switch_name, val) multiple times (Duplication)
OptionParser#parse_in_order calls sw.block multiple times (Duplication)
OptionParser#parse_in_order calls sw.switch_name multiple times (Duplication)
OptionParser#parse_in_order has approx 30 statements (Long Method)
OptionParser#parse_in_order/block is controlled by argument setter (Control Couple)
OptionParser#parse_in_order/block/block is nested (Nested Iterators)
OptionParser#parse_in_order/block/block/block is nested (Nested Iterators)
OptionParser#permute calls argv[0] multiple times (Duplication)
OptionParser#permute refers to argv more than self (Feature Envy)
OptionParser#search/block has the variable name 'k' (Uncommunicative Name)
OptionParser#summarize has 4 parameters (Long Parameter List)
OptionParser#summarize/block has the variable name 'l' (Uncommunicative Name)
OptionParser#ver has the variable name 'v' (Uncommunicative Name)
block has the variable name 'q' (Uncommunicative Name)
